Assets/Scripts/Rocket/NewRocketControl.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class NewRocketControl : MonoBehaviour
+{
+    public float thrusterInput;
+    private float steeringInput;
+    private float rotationAngle;
+
+    //private float vernierFuel = 100f;
+
+    public float thrustMultiplier;
+    public float reverseThrustMultiplier;
+    public float vernierThrusterMultiplier;
+
+    private float rocketUpVelocity;
+    private float rocketRightVelocity;
+
+    public float rocketVerticalMaxSpeed;
+    public float rocketHorizontalMaxSpeed;
+
+    public float rotationSpeed;
+
+    public float rocketAngleValue;
+
+    public float dragValue;
+    public float dragTime;
+
+    [SerializeField] private Rigidbody2D rocketRigidBody2D;
+
+    [SerializeField] private LandingGearController landingGearController;
+
+    void Update()
+    {
+        SetInput();
+        ApplyRocketThrust();
+        ApplyRocketSteering();
+        ApplyRocketVernierThruster();
+        UpdateScreenRocketAngle();
+    }
+
+    private void ApplyRocketThrust()
+    {
+        rocketUpVelocity = Vector2.Dot(transform.up, rocketRigidBody2D.velocity);
+        if (rocketUpVelocity > rocketVerticalMaxSpeed && thrusterInput > 0)
+        {
+            return;
+        }
+        if (rocketRightVelocity > rocketHorizontalMaxSpeed && thrusterInput > 0)
```

Comment:
don't you need to handle right and left? Also, you could probably use Mathf.Clamp instead.